src/script/conversation/MessageRepository.ts

```diff
@@ -602,7 +613,47 @@ export class MessageRepository {
     return this.eventService.updateEventAsUploadFailed(message_et.primary_key, reason);
   }
 
-  private async sendAssetRemotedata(conversation: Conversation, file: Blob, messageId: string, asImage: boolean) {
+  /**
+   * Create asset metadata message to specified conversation.
+   */
+  private async createAssetMetadata(
+    conversation: Conversation,
+    file: File | Blob,
+    allowImageDetection?: boolean,
+    originalId?: string,
+  ) {
+    let metadata;
+    try {
+      metadata = await buildMetadata(file);
+    } catch (error) {
+      const logMessage = `Couldn't render asset preview from metadata. Asset might be corrupt: ${
+        (error as Error).message
+      }`;
+      this.logger.warn(logMessage, error);
+    }
+
+    const meta = {length: file.size, name: (file as File).name, type: file.type} as Partial<FileMetaDataContent>;
+
+    if (isAudio(file)) {
+      meta.audio = metadata as AudioMetaData;
+    } else if (isVideo(file)) {
+      meta.video = metadata as VideoMetaData;
+    } else if (allowImageDetection && isImage(file)) {
+      meta.image = metadata as ImageMetaData;
+    }
+
+    const message = MessageBuilder.buildFileMetaDataMessage({metaData: meta as FileMetaDataContent}, originalId);
+    this.assetRepository.addToProcessQueue(message, conversation.id);
+    return {message, metaData: meta as FileMetaDataContent};
```

Comment:
(minor) We could refactor it a little:

- we could wrap the whole code inside the try/catch block, avoiding reassigning the metadata variable
- is the `Partical` required here? I think we could omit it and just use the`FileMetaDataContent`, because we would have all required variables (based on `packages/core/src/conversation/content/FileContent.ts`)
- by that, we could also create the meta object without dynamically adding properties to it, perform a condition, and if it's true, assign the `metadata`, otherwise leave it `undefined`

```suggestion
    try {
    const metadata = await buildMetadata(file);
    
    const meta = {
      length: file.size, 
      name: (file as File).name, 
      type: file.type,
      audio: isAudio(file) && metadata,
      video: isVideo(file) && metadata,
      image allowImageDetection && isImage(file) && metadata
     } as FileMetaDataContent;

    const message = MessageBuilder.buildFileMetaDataMessage({metaData: meta}, originalId);
    
    this.assetRepository.addToProcessQueue(message, conversation.id);
    
    return {message, metaData: meta};
    } catch (error) {
      const logMessage = `Couldn't render asset preview from metadata. Asset might be corrupt: ${
        (error as Error).message
      }`;
      this.logger.warn(logMessage, error);
    }
```